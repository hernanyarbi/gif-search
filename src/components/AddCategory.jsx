import PropTypes from "prop-types";
import { useState } from "react";

export const AddCategory = ({onAddCategory}) => {

  const [categoryInput, setCategoryInput] = useState('')

  const handleCategoryInput = (event) => {
    const newCategory = event.target.value
    setCategoryInput(newCategory)
  }

  const onAddCategoryI = (event) => {
    event.preventDefault()
    if (!categoryInput.length) return
    onAddCategory(categoryInput)
    setCategoryInput('')
  }

  return (
    <form onSubmit={onAddCategoryI} aria-label='add-category-form'>
      <input type='text' value={categoryInput} onChange={handleCategoryInput}/>
    </form>
  )
}

AddCategory.propTypes = {
  onAddCategory: PropTypes.func.isRequired
}