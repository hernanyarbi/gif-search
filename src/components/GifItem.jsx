
export const GifItem = ({ title, url }) => {
  return (

    <div className="card">
      <img name={title} src={url} />
      <p>{title}</p>
    </div>
  )
}
