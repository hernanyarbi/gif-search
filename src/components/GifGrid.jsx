import PropTypes from "prop-types";
import { useFetchGif } from "../hooks/useFetchGif";
import { GifItem } from "./GifItem";

export const GifGrid = ({ category }) => {

  const { gifs, isLoading } = useFetchGif(category)
  return (
    <>
      <h1>{category}</h1>
      {
        isLoading
          ? (<>cargando...</>)
          : (
            <div className="card-grid">
              {gifs.map(gif => (
                <GifItem key={gif.id} {...gif} />
              ))}
            </div>
          )
      }
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
