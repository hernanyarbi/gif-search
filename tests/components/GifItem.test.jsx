import { render, screen } from "@testing-library/react";
import { GifItem } from "../../src/components/GifItem";

describe('Testing <GifItem/>', () => {
  const title = 'title'
  const url = 'https://test.com/'
  test('should match with stapshot', () => {
    const { container } = render(<GifItem title={title} url={url} />)
    expect(container).toMatchSnapshot()
  });
  
  test('should have a text and alt', () => {
    render(<GifItem title={title} url={url} />)
    const {src, alt} = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  });
  
  test('should have a title', () => {
    render(<GifItem title={title} url={url} />)
    expect(screen.getByText(title)).toBeTruthy()
  });
});