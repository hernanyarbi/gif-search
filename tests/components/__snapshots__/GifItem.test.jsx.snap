// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <GifItem/> should match with stapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="title"
      src="https://test.com/"
    />
    <p>
      title
    </p>
  </div>
</div>
`;
